app/controller/button/StreetViewTool.js

```diff
@@ -0,0 +1,380 @@
+        var overlayLayers;
+        if (overlayGroup) {
+            overlayLayers = overlayGroup.getLayers();
+        }
+
+        if (pressed) {
+            if (overlayLayers) {
+                // add layer and raise layer to top of stack
+                overlayLayers.insertAt(overlayLayers.getLength(), me.vectorLayer);
+            }
+        } else {
+            if (overlayLayers) {
+                overlayLayers.remove(me.vectorLayer);
+            }
+            // cleanup
+            me.vectorLayer.getSource().clear();
+            me.streetViewWin.close();
+        }
+
+        // activate / deactivate click
+        me.registerMapListeners(pressed);
+    },
+
+    /**
+     * Registers listeners on the map we need for this tool.
+     *
+     * @param  {Boolean} activate Flag to activate / deactivate the listeners
+     */
+    registerMapListeners: function(activate) {
+        var me = this;
+
+        if (activate) {
+            me.map.on('singleclick', me.onMapClick, me);
+        } else {
+            me.map.un('singleclick', me.onMapClick, me);
+        }
+    },
+
+    /**
+     * Handles 'singleclick' event on the map.
+     * Show window with Street View panorama for clicked position (if existing)
+     *
+     * @param  {ol.MapBrowserEvent} evt OL event object
+     * @private
+     */
+    onMapClick: function (evt) {
+        var me = this;
+        var clickCoord = evt.coordinate;
+
+        var gmapsLatLng = me.olCoord2GmapsLatLng(clickCoord);
+        me.showStreetViewWindow(gmapsLatLng);
+    },
+
+    /**
+     * Shows the window with the Street View panorama at the given position or
+     * fires an event 'missingpanorama' on the view instance of this controller.
+     *
+     * @param  {google.maps.LatLng} latLng Position to show SV panorama at
+     * @private
+     */
+    showStreetViewWindow: function (latLng) {
+        var me = this;
+        var view = me.getView();
+
+        if (!me.streetViewWin) {
+            me.streetViewWin = Ext.create('Ext.window.Window', {
+                title: view.svWinTitlePrefix,
+                width: view.svWinWidth,
+                height: view.svWinHeight,
+                closeAction: 'destroy',
+                autoShow: false,
+                listeners: {
+                    afterrender: function (win) {
+                        me.svDiv = Ext.getDom(win.getId());
+                    },
+                    destroy: function () {
+                        me.streetViewWin = null;
+                        me.vectorLayer.getSource().clear();
+                    }
+                }
+            });
+        }
+
+        // load the panorama for the given position
+        me.streetViewService.getPanoramaByLocation(latLng, 50, function (data, status) {
+            if (status === google.maps.StreetViewStatus.OK) {
+                var win = me.streetViewWin;
+                var title = view.svWinTitlePrefix + ' Image Date: ' + data.imageDate;
```

Comment:
Probably best to make the text configurable? Probably also for the way you detect the Overlay group by name, above somewhere.